Defra.Cdp.Backend.Api/Services/Deployments/DeploymentsService.cs

```diff
@@ -298,6 +314,108 @@ public async Task<Paginated<Deployment>> FindLatest(string[]? favouriteTeamIds,
+            var teamFilter = builder.In(d => d.Service, servicesOwnedByTeam);
+            filter &= teamFilter;
+            migrationFilter &= builderMigration.In(m => m.Service, servicesOwnedByTeam);
+        }
+
+        if (!string.IsNullOrWhiteSpace(service))
+        {
+            var serviceFilter = builder.Regex(d => d.Service,
+                new BsonRegularExpression(service, "i"));
+            filter &= serviceFilter;
+            migrationFilter &= builderMigration.Regex(m => m.Service, new BsonRegularExpression(service, "i"));;
+        }
+
+        if (!string.IsNullOrWhiteSpace(user))
+        {
+            var userFilter = builder.Where(d =>
+                (d.User != null && d.User.Id == user)
+                || (d.User != null && d.User.DisplayName.ToLower().Contains(user.ToLower())));
+            filter &= userFilter;
+            
+            migrationFilter &= builderMigration.Where(d =>
+                (d.User.Id == user)
+                || (d.User.DisplayName != null && d.User.DisplayName.ToLower().Contains(user.ToLower())));
+        }
+        
+        if (!string.IsNullOrWhiteSpace(status))
+        {
+            var statusLower = status.ToLower();
+            var statusFilter = builder.Where(d =>
+                d.Status.ToLower() == statusLower || d.Status.ToLower().Contains(statusLower));
+            filter &= statusFilter;
+            // Don't filter migration status as they're completely different to deployment status.
+        }
+
+        var migrationCollection = Collection.Database.GetCollection<DatabaseMigration>("migrations");
+        var migrationPipeline = new EmptyPipelineDefinition<DatabaseMigration>()
+            .Match(migrationFilter)
+            .Project(m => new DeploymentOrMigration
+        {
+            Migration = m,
+            Updated = m.Updated
+            
+        });
+
+        var pipeline = new EmptyPipelineDefinition<Deployment>()
+            .Match(filter)
+            .Project(d => new DeploymentOrMigration { Deployment = d, Updated = d.Updated})
+            .UnionWith(migrationCollection, migrationPipeline)
+            .Sort(new SortDefinitionBuilder<DeploymentOrMigration>().Descending(d => d.Updated))
+            .Skip(offset + size * (page - DefaultPage))
+            .Limit(size);
+
+        
+        //var deployments = await Collection.Aggregate(pipeline).ToListAsync(ct);
+        var deployments = await Collection.Aggregate(pipeline).ToListAsync(ct);
+        if (favouriteTeamIds?.Length > 0)
+        {
+            var repos = (await Task.WhenAll(favouriteTeamIds.Select(teamId =>
+                    _repositoryService.FindRepositoriesByTeamId(teamId, true, ct))))
+                .SelectMany(r => r)
+                .ToList();
+
+            var servicesOwnedByTeam = repos.Select(r => r.Id);
+
+            //deployments = deployments.OrderByDescending(d => servicesOwnedByTeam.Contains(d.GetValue("service").AsString)).ToList();
```

Comment:
ditto

Reply:
@christopherjturner Its looking like this is wip? For the portal I would like to bring all deployments and migrations up to the top by passing through `favouriteTeamIds`. Same as we currently are doing on the deployments list page. I'll be looking at integrating this today.

Reply:
I think this just needs fixing up to order by favourit